kafka/src/kafka/impl/configuration.cpp

```diff
@@ -282,6 +291,8 @@ void Configuration::SetOption(const char* option, const char* value, T to_print)
 #pragma GCC diagnostic pop
 #endif
     if (err == RD_KAFKA_CONF_OK) {
+        rd_kafka_conf_set_log_cb(conf_.GetHandle(), KafkaLogCallback);
```

Comment:
IMHO, it is strange place for callback
Why you need to set callback for each SetOption call?
Maybe add callback setting into Configuration constructors?

Reply:
It seems to me that in a good way all logs should fall into one place.
I also moved the callback setting to the configuration constructor.